### ISSUE TITLE
feat(pr-20): Flow-KL regression objective — closed-form Gaussian matching

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#34]
**Part of the Phase 2 epic.**

A regression training objective distinct from maximum-likelihood: push a latent through a flow, then minimise the closed-form Gaussian KL between the flow's output distribution and the target distribution, with **learnable variances on both sides** (variational σ's). Useful when you want to treat regression as "distribution matching" rather than "point prediction + fixed-σ likelihood".

## Reference (prior art)

Three variants in [`code/models/regression.py`](https://github.com/acids-ircam/flow_synthesizer/blob/master/code/models/regression.py):

- `FlowKL`: `KL(N(flow(z), σ_z) || N(y, σ_y))` with two `VariationalLambda` variances
- `FlowKLFull`: adds a prior log-prob on `y` and an explicit Jacobian term
- `FlowMixture` (in `regression.backup.py`): dim-wise learnable variances

## Scope

New module `src/gauss_flows/_src/objectives.py` (new file for reusable loss functions):

```python
def flow_kl_regression_loss(
    flow,                 # conditional bijection
    context: Array,       # x
    target: Array,        # y
    sigma_z: VariationalSigma,
    sigma_y: VariationalSigma,
    *,
    include_jacobian: bool = True,
) -> Float:
    \"\"\"Variational Gaussian-matching loss between flow output and target.\"\"\"
```

- Return value includes the aggregated KL plus variational KL terms from both σ's (so it's a scalar ELBO-style loss)
- Optional `include_jacobian` flag for the `FlowKLFull`-style variant

## Acceptance criteria

- [ ] Matches the closed-form KL between two diagonal Gaussians to 1e-5 (tested against `jax.scipy`)
- [ ] Smoke test: trains a small network to fit a 2D heteroscedastic regression where MSE fails
- [ ] Docstring explains when to prefer this vs log-likelihood regression

## Dependencies

**Blocks**: nothing
**Blocked by**: #32 (VariationalSigma)
